AdvancedDLSupport/NativeLibraryBase.cs

```diff
@@ -51,38 +52,43 @@ public abstract class NativeLibraryBase : IDisposable
         /// Gets the library and symbol loader for the current platform.
         /// </summary>
         [NotNull]
-        private static readonly IPlatformLoader PlatformLoader;
+        private static IPlatformLoader PlatformLoader => PlatformLoaderBase.PlatformLoader;
 
-        /// <summary>
-        /// Initializes static members of the <see cref="NativeLibraryBase"/> class.
-        /// </summary>
-        static NativeLibraryBase()
-        {
-            PlatformLoader = PlatformLoaderBase.SelectPlatformLoader();
-        }
+        private readonly ILibraryLoader _libLoader;
+        private readonly ISymbolLoader _symbolLoader;
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="NativeLibraryBase"/> class.
-        /// </summary>
-        /// <param name="path">The path to the library.</param>
-        /// <param name="options">Whether or not this library can be disposed.</param>
+        private ILibraryLoader LibraryLoader => _libLoader ?? PlatformLoader;
```

Comment:
Remove these two properties, and assign _libraryLoader and _symbolLoader once in the constructor instead.